private/rdf4cpp/rdf/parser/IStreamQuadIteratorSerdImpl.hpp

```diff
@@ -57,6 +57,18 @@ struct IStreamQuadIterator::Impl {
     static SerdStatus on_prefix(void *voided_self, SerdNode const *name, SerdNode const *uri) noexcept;
     static SerdStatus on_stmt(void *voided_self, SerdStatementFlags, SerdNode const *graph, SerdNode const *subj, SerdNode const *pred, SerdNode const *obj, SerdNode const *obj_datatype, SerdNode const *obj_lang) noexcept;
 
+    static constexpr SerdSyntax extract_syntax_from_flags(ParsingFlags flags) noexcept {
+        if (flags.contains(ParsingFlag::Turtle))
```

Comment:
If you implement the thing from below, just take care to check the things in the correct order. Or alternatively make this into a switch on the flag's underlying repr with the non-syntax bits masked out (not that important tbh, you choose which of the options you like better).